woocommerce/woocommerce
# Update Filterer.php to resolve #34977

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Add the `woocommerce_filterer_hide_out_of_stock` filter, used by the product attributes lookup table filterer class. This allows stores to show the out of stock product variations in the catalog but hide them if a user use the sidebar filters through the Layered Nav Widget. Note that this only applies when the "Use the procut attributes lookup table" setting is enabled.

Closes #34977 .



### How to test the changes in this Pull Request:



Code review should be sufficient for this change since it just apply a filter to a value already present in the code base.

1. Enable the usage of the product attributes lookup table (in WooCommerce - Settings - Products - Advanced).
1. Create a test plugin at wp-content/plugins/test-filterer-pr/test-filterer-pr.php;
2. It should contain the following code: 
```php
<?php
   /**
    * Plugin Name: Test Filterer class patch PR
    * Description: Test plugin for PR validation.
    */
   
   add_action( 'woocommerce_filterer_hide_out_of_stock', '__return_true' );
```
3. Ensure that the option to hide out of stock items from the shop is disabled (woocommerce_hide_out_of_stock_items, in WooCommerce - Settings - Products - Inventory);
4. Create one product, assign it any attribute term and make it out of stock;
5. Create one product, assign it the same attribute term and make it available;
6. Add a Layered Nav Widget for the used attribute to the shop sidebar;
7. You should be able to see both product in the shop without filters and see only the available one if you filter for the attribute above.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
New 'woocommerce_filterer_hide_out_of_stock' filter
</details>
